Bring project flow to new client

AdityaKhatri opened the pull request:
## Changes

* Bring project edit flow in new DEEP

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [ ] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

AdityaKhatri committed:
Bring project flow to new client

tnagorra commented on app/Base/configs/routes.tsx:
```diff
@@ -108,6 +108,34 @@ const analyticalFrameworkEditRoute = wrap({
+        return project.allowedPermissions.includes('UPDATE_PROJECT');
+    },
+    */
+});
```
This should be nested inside the project route

tnagorra commented on app/Base/configs/routes.tsx:
```diff
@@ -108,6 +108,34 @@ const analyticalFrameworkEditRoute = wrap({
+    component: lazy(() => import('#views/ProjectEdit')),
+    componentProps: { },
+    visibility: 'is-authenticated',
+    // NOTE: we cannot use permission check related to project on this route
+    // as this route manages all the data load
```
We can just have two route
/projects/12/edit
projects/new

tnagorra commented on app/views/ProjectEdit/Framework/FrameworkDetail/index.tsx:
```diff
@@ -96,7 +96,7 @@ function FrameworkDetail(props: Props) {
+    const itemRendererParams = useCallback((_, data) => ({
         className: styles.projectTitle,
         children: data.title,
         to: '',
```
Can this not be filled yet?

AdityaKhatri replied:
This can be filled, but might not be necessary as we don't have a project-specific page

tnagorra commented on app/views/ProjectEdit/ProjectDetailsForm/index.tsx:
```diff
@@ -258,19 +247,17 @@ function ProjectDetailsForm(props: Props) {
+                const projectEditPath = generatePath(routes.projectEdit.path, {
+                    projectId,
+                });
+                history.push(projectEditPath);
```
Let's do history.replace

tnagorra commented on app/views/ProjectEdit/ProjectDetailsForm/index.tsx:
```diff
@@ -283,12 +270,8 @@ function ProjectDetailsForm(props: Props) {
-                ),
-            );
+            const homePath = generatePath(routes.home.path, {});
+            history.push(homePath);
```
Let's do history.replace

tnagorra commented on app/views/ProjectEdit/index.tsx:
```diff
@@ -11,65 +12,44 @@ import {
-        const newRoute = reverseRoute(pathNames.editProject, {
-            redirectId,
+        const newRoute = generatePath(routes.projectEdit.path, {
+            projectId,
```
While redirecting we may need to use replace

tnagorra committed:
Add feature to skip project permission check for other project links

- Use useParams to get projectId for ProjectEdit component
- Create SmartButtonLikeLink component
- Create SmartLink component

tnagorra committed:
Set mapbox token

- Update deep ui
- Fix case when requestError is undefined
